st2common/st2common/models/db/execution.py

```diff
@@ -61,29 +51,39 @@ class ActionExecutionDB(stormbase.StormFoundationDB):
     end_timestamp = ComplexDateTimeField(
         help_text="The timestamp when the liveaction has finished."
     )
-    parameters = stormbase.EscapedDynamicField(
+    action = stormbase.EscapedDictField(required=True)
+    parameters = JSONDictEscapedFieldCompatibilityField(
         default={},
         help_text="The key-value pairs passed as to the action runner & action.",
     )
     result = JSONDictEscapedFieldCompatibilityField(
         default={}, help_text="Action defined result."
     )
-    result_size = me.IntField(default=0, help_text="Serialized result size in bytes")
     context = me.DictField(
         default={}, help_text="Contextual information on the action execution."
     )
+    delay = me.IntField(min_value=0)
+
+    # diff from liveaction
+    runner = stormbase.EscapedDictField(required=True)
+    trigger = stormbase.EscapedDictField()
+    trigger_type = stormbase.EscapedDictField()
+    trigger_instance = stormbase.EscapedDictField()
+    rule = stormbase.EscapedDictField()
+    result_size = me.IntField(default=0, help_text="Serialized result size in bytes")
     parent = me.StringField()
     children = me.ListField(field=me.StringField())
     log = me.ListField(field=me.DictField())
-    delay = me.IntField(min_value=0)
     # Do not use URLField for web_url. If host doesn't have FQDN set, URLField validation blows.
     web_url = me.StringField(required=False)
 
+    # liveaction id
+    liveaction = me.StringField()
+
     meta = {
         "indexes": [
             {"fields": ["rule.ref"]},
             {"fields": ["action.ref"]},
-            {"fields": ["liveaction.id"]},
```

Comment:
todo: add liveaction as an indexed field

Reply:
set liveaction to an index field

Reply:
> When adding new fields or new indexes, MongoEngine will take care of creating them (unless auto_create_index is disabled)

So yes mongo will create the index, but it isn't clear that the old history with the full document will be valid.

Reply:
I suppose changing the field to `liveaction_id` could ensure we don't have issues with the migration of this field.